Rename 'slave' to 'replica' everywhere

aks opened the pull request:
Tsukasa-san,

I hope you do not mind this PR.  Because of increased sensitivity to slavery in American history, there has been some suggestions, even in the technology world, to use other words, such as "replica", in order to avoid the social issues involved with the word "slave".

So, this PR is a change of the fresh_connection code base to use the word "replica", or "replica1", "replica2", etc. instead of "slave".  The change has been made throughout the code base, everywhere. 

I have confirmed that all tests still pass, for both MySQL and PG.

It includes some backward compatibility so that the new gem can continue working with the existing database configurations using the "slave" stanza.  I have included a test to confirm this compatibility.

In addition, I also improved the content and English in the README.  Even if you choose to not accept the "replica" change, there is a lot of good improvement in the README that you may wish to consider _(perhaps I should have accomplished this in a separate PR)_.

Thanks for the useful software; I'm still evaluating it for use my company's application.

Thanks
Alan

tsukasaoishi commented:
@aks 
Thank you for the Pull Request.
I respect and consent all your opinion.
I will merge this PR after reviewing code. Please wait a while.
Thank you to improve README!

aks commented:
Tsukasa-san, the CI process is failing because of either DB existence (new names?) or access control (to the new DB names).

tsukasaoishi commented on test/config/database_mysql2.yml:
```diff
@@ -4,11 +4,11 @@ test:
   pool: 5
   username: root
-  password:
+  password: local-db
```
Password of TravisCI's MySQL is blank.

tsukasaoishi replied:
Password of TravisCI's MySQL is blank.

tsukasaoishi commented on lib/fresh_connection/version.rb:
```diff
@@ -1,4 +1,4 @@
 module FreshConnection
-  VERSION = "2.2.0"
+  VERSION = "2.3.0"
```
I will do version control. Could you revert this change?

aks commented:
Tsukasa-san, I replaced hard-coded user/pass on mysql DB config with envars, but using the same defaults as before this PR.  This makes the test work for me, because I can set `MYSQL_USER` and `MYSQL_PASS` appropriately for my testing, but it should also work for CI because it will have the same defaults when those envars are not defined.

I reverted the version change.

tsukasaoishi commented:
@aks 
Thank you for your contribution!
